app.py

```diff
@@ -0,0 +1,139 @@
+from flask import Flask, request, jsonify, make_response
+from flask_sqlalchemy import SQLAlchemy
+from os import environ
+from abc import ABC, abstractmethod
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = environ.get('DATABASE_URL')
+db = SQLAlchemy(app)
+
+
+class User(db.Model):
+    __tablename__ = 'users'
+
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(30), unique=False, nullable=False)
+    email = db.Column(db.String(50), unique=True, nullable=False)
+
+    def json(self):
+        return {
+            'id': self.id,
+            'username': self.username,
+            'email': self.email
+        }
+
+
+class AbstractDatabaseAdapter(ABC):
+
+    @abstractmethod
+    def create(self, entity):
+        pass
+
+    @abstractmethod
+    def get_all(self, entity_type):
+        pass
+
+    @abstractmethod
+    def get_by_id(self, entity_type, entity_id):
+        pass
+
+    @abstractmethod
+    def update(self, entity):
+        pass
+
+    @abstractmethod
+    def delete(self, entity):
+        pass
+
+
+class PostgreSQLAdapter(AbstractDatabaseAdapter):
+
+    def create(self, entity):
+        db.session.add(entity)
+        db.session.commit()
+
+    def get_all(self, entity_type):
+        return entity_type.query.all()
+
+    def get_by_id(self, entity_type, entity_id):
+        return entity_type.query.get(entity_id)
+
+    def update(self, entity):
+        db.session.commit()
+
+    def delete(self, entity):
+        db.session.delete(entity)
+        db.session.commit()
+
+
+db_adapter = PostgreSQLAdapter()
+
+with app.app_context():
```

Comment:
I'm not really that familiar with SQLAlchemy but can't it be done on db container start? Like django's migrations?

Reply:
there is such a thing for flask as Flask-Migrate, but for the moment I don't know exactly how to use it, so I used another solution that works :)